pkg/apiserver/util/openapi.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"bytes"
+	"github.com/emicklei/go-restful"
+	"strings"
+	"unicode"
+)
+
+func createValidName(s string, capitalizeFirstLetter bool) string {
```

Comment:
This is not only camel casing, it is making a valid variable name, so for example, it removes punctuations, whitespaces, etc. what about toValidOpId with a doc line?